test/expr/cast/literals_downcast.swift

```diff
@@ -2,17 +2,17 @@
 
 let ok = "A" as Character // OK
 let succeed = "A" as? String // expected-warning {{always succeeds}}
-let bad = "A" as? Character // expected-warning {{conditional downcast from literal to 'Character' always fails; consider using 'as' coercion}} {{none}}
+let bad = "A" as? Character // expected-warning {{conditional downcast from literal of inferred type 'String' to unrelated type 'Character' always fails; consider using 'as' coercion}} {{none}}
```

Comment:
Can you add a test with `as!` too? It’s not necessarily a *conditional* downcast, so I think we should remove that elaboration from the message. It’s not necessarily even a downcast for that matter, but that’s not that big of a deal.

Reply:
Sure!

Reply:
Could you tell me the exact `expected-warning` that must be added for the `as!` case?

Reply:
Have you tried running the test? It should print unexpected or incorrect warnings.

Reply:
I have made the changes.

Reply:
As I mentioned earlier, we use the same error message for `as?` and `as!`, but the message says "conditional downcast". This is a good opportunity to get rid of this inaccuracy if you’re interested.